fix nasa#2316 - adding CFE_TIME_StringFmt() and CFE_TIME_StringFmtLen…

CDKnightNASA opened the pull request:
…() functions

**Checklist (Please check before submitting)**

* [X] I reviewed the [Contributing Guide](https://github.com/nasa/cFE/blob/main/CONTRIBUTING.md).
* [X] I signed and emailed the appropriate [Contributor License Agreement](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md#contributor-license-agreement-cla) to GSFC-SoftwareRelease@mail.nasa.gov and copied cfs-program@lists.nasa.gov.

**Describe the contribution**
Fixes #2316, adds CFE_TIME_StringFmt() and CFE_TIME_StringFmtLen() functions that behave similarly to strftime (but limited to %Y [yyyy], %j [doy], %H, %M, %S, %f [micro]). CFE_TIME_Print() calls CFE_TIME_StringFmt().

**Testing performed**
Added UT tests, all pass.

**Expected behavior changes**
No impact to behavior, adds capabilities (that will be extended to EVS so that log message timestamps can be configurable.)

**System(s) tested on**
 - Hardware: PC
 - OS: Debian "Bookworm" (12)
 - Versions: HEAD of main branch of cFE

**Additional context**
An alternative would be to use strftime, if all target platforms include it. This would allow for month/day-of-month and other options this code does not include and would be difficult to implement.

**Third party code**
None

**Contributor Info - All information REQUIRED for consideration of pull request**
Christopher.D.Knight@nasa.gov

skliper commented:
Consider returning errors and remove low level `CFE_ES_WriteToSysLog` use so the calling function can report issues to avoid recursion if/where possible.  This is another case where it would be nice to have a health parameter in tlm vs relying on syslog. 
- #1469

CDKnightNASA commented:
> Consider returning errors and remove low level `CFE_ES_WriteToSysLog` use so the calling function can report issues to avoid recursion if/where possible. This is another case where it would be nice to have a health parameter in tlm vs relying on syslog.

Copy on the WriteToSysLog, latest revs removes. (Automated tests flagged this as well, nice!) All of the new automated checks are making me do more work. :) Will take out of draft once I've addressed all issues.
